mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -1020,6 +1043,74 @@ impl TendermintCoin {
         let min_tx_amount = big_decimal_from_sat(MIN_TX_SATOSHIS, decimals);
         amount >= &min_tx_amount
     }
+
+    async fn search_for_swap_tx_spend(
+        &self,
+        input: SearchForSwapTxSpendInput<'_>,
+    ) -> MmResult<Option<FoundSwapTxSpend>, SearchForSwapTxSpendErr> {
+        let tx = cosmrs::Tx::from_bytes(input.tx)?;
+        let first_message = tx
+            .body
+            .messages
+            .first()
+            .or_mm_err(|| SearchForSwapTxSpendErr::TxMessagesEmpty)?;
+        let htlc_proto = CreateHtlcProtoRep::decode(first_message.value.as_slice())?;
+        let htlc = MsgCreateHtlc::try_from(htlc_proto)?;
+        let htlc_id = self.calculate_htlc_id(&htlc.sender, &htlc.to, htlc.amount, input.secret_hash);
+
+        let htlc_response = self.query_htlc(htlc_id.clone()).await?;
+        let htlc_data = match htlc_response.htlc {
+            Some(htlc) => htlc,
+            None => return Ok(None),
+        };
+
+        match htlc_data.state {
+            HTLC_STATE_OPEN => Ok(None),
+            HTLC_STATE_COMPLETED => {
+                let events_string = format!("claim_htlc.id='{}'", htlc_id);
+                let request = GetTxsEventRequest {
+                    events: vec![events_string],
+                    pagination: None,
+                    order_by: 0,
```

Comment:
`TendermintResultOrder::Ascending` can be used here.

Reply:
@ozkanonur I got `expected i32, found enum cosmrs::tendermint_rpc::Order` error when tried to use it. `TendermintResultOrder` is used in `tx_search` and `block_search`, but not for `GetTxsEventRequest`.

Reply:
> TendermintResultOrder::Ascending

I thught they implemented `into` for numbers. I will create new `Order` enum which implements `TendermintResultOrder` type and also can convert to `i32`. Since this PR is P0 I will do this refactoring on my side.